### ISSUE TITLE
Metrics reporter failed to collect CPU metric

### ISSUE THREAD

**Issue by lkalina-cera** [linkedin/cruise-control#1912]
Hi team,
In the recent days we faced an issue with the cruise control metric reporter on one of the kafka brokers.
The exception
```2022-09-16 14:24:18,424 WARN com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter: [CruiseControlMetricsReporterRunner]: Failed reporting CPU util. java.io.IOException: Java Virtual Machine recent CPU usage is not available. 
at com.linkedin.kafka.cruisecontrol.metricsreporter.metric.MetricsUtils.getCpuMetric(MetricsUtils.java:414) ~[cruise-control-metrics-reporter-2.5.85.7.2.16.0-171.jar:?] 
at com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter.reportCpuUtils(CruiseControlMetricsReporter.java:449) [cruise-control-metrics-reporter-2.5.85.7.2.16.0-171.jar:?]
at com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter.run(CruiseControlMetricsReporter.java:367) [cruise-control-metrics-reporter-2.5.85.7.2.16.0-171.jar:?] 
at java.lang.Thread.run(Thread.java:750) [?:1.8.0_342]
```
After a long journey it turned out that the `com.sun.management.OperatingSystemMXBean.getProcessCpuLoad()` returns with -1 on this machine. This CPU metric reporting works fine on the other brokers (same operating system, same installation) the only difference what I found is the CPU.
Working brokers have these CPUs Intel(R) Xeon(R) CPU E5-2630 v3 @ 2.40GHz ,Intel(R) Xeon(R) Silver 4215R CPU @ 3.20GHz (32 cpus). The non working one has Intel(R) Xeon(R) CPU E5-2699 v4 @ 2.20GHz (88 cpus)
To sum up if one broker could not report this CPU metric the NumValidWindows stuck at 0 and proposal generation and rebalance will not work.
I tested `this com.sun.management.OperatingSystemMXBean. getProcessCpuLoad` method on the wrong broker with many JDK versions and it works from Java 19. But unfortunately this is not an option for us to move or stuff from Java 8/ Java 11 to 19.
Can you fix this issue? Or can you implement a workaround if the getProcessCpuLoad returning with -1?
I have already raised it on bugreport.java.com and I have an internal review ID but it might take a long time to fix and backport it
CC:   @kyguy @efeg

**Comment by efeg** [comment]
I think the right long-term fix should indeed be a backport on earlier JDK versions.

-- In the meanwhile, there are a few ways to address the issue. For example, you can create a new metrics reporter class that extends `CruiseControlMetricsReporter` and override `reportCpuUtils` (it needs to be changed to `protected`) to handle `IOException` and return some other value.
